### PR TITLE
KDESKTOP-1346-Fail-to-retrieve-the-public-share-link-on-2nd-call

### PR DESCRIPTION
**Pull request by ClementKunz**
Fix regression on get file share link

### CODE REVIEW

File: src/libsyncengine/requests/serverrequests.cpp [hunk 1]
```diff
@@ -1067,10 +1067,8 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
         return ExitCode::DataError;
     }
 
-    ExitCode exitCode = job->runSynchronously();
-    if (exitCode != ExitCode::Ok) {
-        std::string errorCode;
-        if (job->hasErrorApi(&errorCode) && getNetworkErrorCode(errorCode) == NetworkErrorCode::fileShareLinkAlreadyExists) {
+    if (auto exitCode = job->runSynchronously(); exitCode != ExitCode::Ok) {
```
[the remark this answers is not in the document]
Comment (ClementKunz) [reply src/libsyncengine/requests/serverrequests.cpp]: Yes, you are right. I refactored the whole function in 74ec54e422d6d058eaf4ed1acc0546b970a03036

**Review by herve-er — APPROVED** [review]

File: src/libsyncengine/requests/serverrequests.cpp [hunk 2]
```diff
@@ -1067,63 +1067,41 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
         return ExitCode::DataError;
     }
 
-    if (auto exitCode = job->runSynchronously(); exitCode != ExitCode::Ok) {
+    if (job->runSynchronously() != ExitCode::Ok) {
         if (job->exitCode() == ExitCode::BackError && job->exitCause() == ExitCause::ShareLinkAlreadyExist) {
-            // Get link
-            std::shared_ptr<GetFileLinkJob> job2;
+            // The link already exist, get it
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/requests/serverrequests.cpp:1072]:
```suggestion
            // The link already exists, get it
```

File: src/libsyncengine/requests/serverrequests.cpp [hunk 3]
```diff
@@ -1067,63 +1067,41 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
         return ExitCode::DataError;
     }
 
-    if (auto exitCode = job->runSynchronously(); exitCode != ExitCode::Ok) {
+    if (job->runSynchronously() != ExitCode::Ok) {
         if (job->exitCode() == ExitCode::BackError && job->exitCause() == ExitCause::ShareLinkAlreadyExist) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/requests/serverrequests.cpp:1071]:
```suggestion
        if (job->exitCode() == ExitCode::BackError && job->exitCause() == ExitCause::ShareLinkAlreadyExists) {
```

If it does not cause too many replacements.

File: src/libsyncengine/requests/serverrequests.cpp [hunk 4]
```diff
@@ -1067,63 +1067,41 @@ ExitCode ServerRequests::getPublicLinkUrl(int driveDbId, const QString &fileId,
...
             }
-            linkUrl = QString::fromStdString(tmp);
-
-            return ExitCode::Ok;
         } else {
-            LOG_WARN(Log::instance()->getLogger(), "Error in PostFileLinkJob::PostFileLinkJob for driveDbId="
-                                                           << driveDbId << " nodeId=" << nodeId.c_str() << " code=" << exitCode);
-            return exitCode;
+            LOG_WARN(Log::instance()->getLogger(),
+                     "Error in PostFileLinkJob::PostFileLinkJob for driveDbId=" << driveDbId << " nodeId=" << nodeId.c_str()
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/requests/serverrequests.cpp:1091]: **Suggestion.** Use a lambda to generate the log message. The same message is built at least 3 times.

File: src/libsyncengine/requests/serverrequests.cpp (hunk 2 above)
Comment (ClementKunz) [reply src/libsyncengine/requests/serverrequests.cpp]: Done in 3479496b5a2452349bf511bb620f292b9214faba

File: src/libsyncengine/requests/serverrequests.cpp (hunk 4 above)
Comment (ClementKunz) [reply src/libsyncengine/requests/serverrequests.cpp]: Done in 3479496b5a2452349bf511bb620f292b9214faba

**Review by luc-guyot-infomaniak — APPROVED** [review]

**Pull request merged**